src/ingest_validation_tools/directory_validator.py

```diff
@@ -0,0 +1,96 @@
+from os import walk
+from fnmatch import fnmatch
+
+
+class DirectoryValidationErrors(Exception):
+    def __init__(self, errors):
+        self.errors = errors
+
+
+def validate_directory(path, paths_dict, dataset_ignore_globs=[]):
+    '''
+    Given a directory path, and a dict representing valid path globs,
+    raise DirectoryValidationErrors if there are errors.
+    '''
+    required_globs, allowed_globs = _get_required_allowed(paths_dict)
+    required_missing_errors, not_allowed_errors = ([], [])
+    if not path.exists():
+        raise DirectoryValidationErrors({
+            'No such file or directory': str(path)
+        })
+    actual_paths = []
+    for triple in walk(path):
+        (dir_path, dir_names, file_names) = triple
```

Comment:
`_` for `dir_names` since it is not used?

Reply:
I'll add an underscore: It might be useful in the future, and I'd never remember what's in that triple otherwise. Thanks!